great_expectations/execution_engine/partition_and_sample/sqlalchemy_data_partitioner.py

```diff
@@ -300,7 +300,7 @@ def partition_on_hashed_column(
         """Partition on the hashed value of the named column"""
         if self._dialect == GXSqlDialect.SQLITE:
             return (
-                sa.func.md5(sa.cast(sa.column(column_name), sa.VARCHAR), hash_digits)
+                sa.func.sha256(sa.cast(sa.column(column_name), sa.VARCHAR), hash_digits)
```

Comment:
What's the relationship of this hashing and the `batch_identifiers[column_name]`? How is the hashed batch identifier specified?